### PR TITLE
Admin merchants index status

### PR DESCRIPTION
**Pull request by JackCSweeney**
This branch adds the following:
- Adds a migration to update the merchants table to include a column for status with options of enabled and disabled, set up as an enum, default value is set to disabled
- Adds status attribute to factorybot file
- Adds #enabled? to merchant model
- Adds the ability to enable or disable merchants from the admin merchants index page
   - Merchants are displayed in two separate section based on their status
   - Buttons to change that status are displayed next to their names
   - Buttons go to admin/merchants_controller with params to change the status, update now has conditions on how it updates/redirects based on those params
- Adds link to create new merchants from admin merchants index page
- Adds new and create actions to admin/merchants_controller

All tests are passing

### CODE REVIEW

**Comment by laurarvegav** [comment]
Looks solid, the merge conflict is only related to the add on in Marchant model for the column status.
We'll now do some sad path testing for the forms :)

File: app/controllers/admin/merchants_controller.rb [hunk 1]
```diff
@@ -7,4 +7,38 @@ def index
...
+  end
+  
+  def new
+  end
+
+  def create
+    merchant = Merchant.new(admin_merchant_params)
+    merchant.save
+    redirect_to admin_merchants_path
+  end
```
Comment (laurarvegav) [inline app/controllers/admin/merchants_controller.rb:38]: Yes! for the sad_path I believe we could add something similar to this in the controller:
```
if merchant.save
      redirect_to admin_merchants_path
    else
      redirect_to edit_admin_merchant_path
      flash[:alert] = "Error: #{error_message(merchant.errors)}"
    end
```

Comment (JackCSweeney) [reply app/controllers/admin/merchants_controller.rb:38]: Perfect! Yeah, we can get this sad path set up for both that create and edit section for the admin merchants!

Comment (laurarvegav) [reply app/controllers/admin/merchants_controller.rb:38]: Also then in the views/layouts/application.html.erb

`<% flash.each do |type, msg| %>
    <div>
      <%= msg %>
    </div>`

**Pull request merged**